app/main.py

```diff
@@ -1,13 +1,33 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "errors": [format_linter_error(erroy) for erroy in errors],
+        "path": file_path,
+        "status": "failed" if len(errors) > 0 else "passed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        {
+            "errors": [
+                format_linter_error(error)
+                for error
+                in linter_report[error_report]
+            ],
+            "path": error_report,
+            "status": "failed"
+                      if len(linter_report[error_report]) > 0
+                      else "passed"
+        }
+        for error_report in linter_report
+    ]
```

Comment:
Call `format_single_linter_file` funtion here and iterate through `linter_report` using `.items()` to not retrieve value by key but rather retrieve it from the beginning